### PR TITLE
added Karaf feature definitions

### PR DESCRIPTION
**Pull request by kaikreuzer**
Also-By: Markus Rathgeb <maggu2810@gmail.com>
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: features/karaf/pom.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <groupId>org.eclipse.smarthome</groupId>
+    <artifactId>features</artifactId>
+    <version>0.8.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>karaf</artifactId>
+  <packaging>feature</packaging>
+
+  <properties>
+  	<karaf.version>4.0.3</karaf.version>
```
Comment (maggu2810) [inline features/karaf/pom.xml]: At least this line is using mixed tab(s) and space(s) for indentation.

Comment (maggu2810) [inline features/karaf/pom.xml]: The bundle org.eclipse.smarthome.io.console.karaf is using "4.0.0.M2".
Shouldn't we define one global karaf version in the root pom (so moving this property to the root one) and use the one property in both artifacts?

Comment (kaikreuzer) [inline features/karaf/pom.xml]: good points. I have updated the PR accordingly.

**Comment by maggu2810** [comment]
Perhaps we could change the artifact name to fit to the remaining ones?

```text
[INFO] Eclipse SmartHome Features ......................... SUCCESS [  0.005 s]
[INFO] Eclipse SmartHome Runtime Core Feature ............. SUCCESS [  0.343 s]
[INFO] Eclipse SmartHome Runtime Model Feature ............ SUCCESS [  0.204 s]
[INFO] Eclipse SmartHome Runtime UI Feature ............... SUCCESS [  0.144 s]
[INFO] Eclipse SmartHome Runtime REST Feature ............. SUCCESS [  0.112 s]
[INFO] Eclipse SmartHome Runtime Automation Feature ....... SUCCESS [  0.146 s]
[INFO] Eclipse SmartHome Runtime Equinox Console Feature .. SUCCESS [  0.049 s]
[INFO] Eclipse SmartHome Runtime RfC147 Console Feature ... SUCCESS [  0.046 s]
[INFO] Eclipse SmartHome Runtime Binding Feature .......... SUCCESS [  0.077 s]
[INFO] Eclipse SmartHome Designer Dependencies ............ SUCCESS [  0.131 s]
[INFO] Eclipse SmartHome Designer Feature ................. SUCCESS [  0.264 s]
[INFO] Eclipse SmartHome Test Dependencies ................ SUCCESS [  0.056 s]
```
```text
[INFO] karaf .............................................. SUCCESS [  3.829 s]
```
```text
[INFO] Eclipse SmartHome Products ......................... SUCCESS [  0.006 s]
[INFO] Eclipse SmartHome Designer RCP Application ......... SUCCESS [ 53.100 s]
[INFO] Eclipse SmartHome p2 Repository .................... SUCCESS [  3.596 s]
[INFO] Eclipse SmartHome Tools ............................ SUCCESS [  0.039 s]
[INFO] Eclipse SmartHome Archetypes ....................... SUCCESS [  0.008 s]
[INFO] Eclipse Smarthome Binding Archetype ................ SUCCESS [  2.481 s]
[INFO] Eclipse Smarthome Binding Archetype Testsuite ...... SUCCESS [  0.038 s]
[INFO] Eclipse SmartHome Target Platform .................. SUCCESS [  0.072 s]
[INFO] Eclipse SmartHome Distribution ..................... SUCCESS [ 27.313 s]
```

**Pull request merged**